File: Question-2/solution.cpp
#include <algorithm>
#include <cstdio>

using namespace std;

int main()
{
   char k1[31], k2[31], rez[61];
   int n1, n2, t, P, i;

   scanf("%d%d", &n1, &n2);
   scanf("%s%s", k1, k2);
   scanf("%d", &t);

   for(i=0;i<n1;i++)
   {
      P = n1-i-1;
      P += min( max(t-i,0), n2 );
      rez[P] = k1[i];
   }
   for(i=0;i<n2;i++)
   {
      P = n1+i;
      P -= min( max(t-i,0), n1 );
      rez[P] = k2[i];
   }
   rez[n1+n2] = '\0';
   printf("%s\n",rez);

   return 0;
}

File: Question-2/solution.java
import java.util.Scanner;

public class Main {

		public static void main(String args[]) {
		Scanner scn=new Scanner(System.in);
		int onelength=scn.nextInt();
		int twolength=scn.nextInt();
		String one=scn.next();
		String two=scn.next();
		int t=scn.nextInt();
		char[] onestr_array=one.toCharArray();
		char[] twostr_array=two.toCharArray();

		char[] thirdarray=new char[onelength+twolength];
		int P=0;
		for(int i=0; i<one.length(); i++) {
			P=onelength-i-1;
			P+=Math.min(Math.max(t-i, 0), twolength);
			thirdarray[P]=onestr_array[i];
		}
		for(int i=0; i<two.length(); i++) {
			P=onelength+i;
			P-=Math.min(Math.max(t-i, 0), onelength);
			thirdarray[P]=twostr_array[i];
		}
		
		for(char ch: thirdarray) {
			System.out.print(ch);
		}
		
	}
}